### PR TITLE
Add Public catalog

### LINKED ISSUE #122: Implement PublicCatalog

### PR DESCRIPTION
**Pull request by FredericBerot-Armand**
closes #122

### CODE REVIEW

File: outscale/data_source_outscale_public_catalog.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+func dataSourceOutscaleOAPIPublicCatalogRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*OutscaleClient).OSCAPI
+
+	req := oscgo.ReadCatalogRequest{}
+
+	var resp oscgo.ReadCatalogResponse
+	var err error
+	err = resource.Retry(20*time.Second, func() *resource.RetryError {
+		resp, _, err = conn.CatalogApi.ReadCatalog(context.Background()).ReadCatalogRequest(req).Execute()
```
Comment (outscale-toa) [inline outscale/data_source_outscale_public_catalog.go:82]: Same here with ```conn.CatalogApi.ReadCatalog```

File: outscale/data_source_outscale_public_catalog.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	"time"
+
+	oscgo "github.com/outscale/osc-sdk-go/v2"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-outscale/utils"
+)
+
+func dataSourceOutscaleOAPIPublicCatalog() *schema.Resource {
```
Comment (outscale-toa) [inline outscale/data_source_outscale_public_catalog.go:14]: ```ReadPublicCatalog``` return too many elements to be a singular data source.
Can you rename  ```dataSourceOutscaleOAPIPublicCatalog```  to ``` dataSourceOutscaleOAPIPublicCatalogs ```and the filename to  ```data_source_outscale_public_catalogs.go``` please?

**Review by outscale-toa — CHANGES_REQUESTED** [review]

File: outscale/data_source_outscale_public_catalog.go (hunk 2 above)
Comment (FredericBerot-Armand) [reply outscale/data_source_outscale_public_catalog.go:14]: It is only one catalog, with many entries no ?

File: outscale/data_source_outscale_public_catalog.go (hunk 1 above)
Comment (FredericBerot-Armand) [reply outscale/data_source_outscale_public_catalog.go]: Yes, my bad too ^^

**Review by outscale-toa — APPROVED** [review]

**Pull request merged**